101-synapse-poc/azuredeploy.json

```diff
@@ -0,0 +1,621 @@
+                "[variables('workspaceName')]"
+            ],
+            "properties": {
+                "mode": "Incremental",
+                "template": {
+                    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+                    "contentVersion": "1.0.0.0",
+                    "resources": [
+                        {
+                            "type": "Microsoft.Storage/storageAccounts/providers/roleAssignments",
+                            "apiVersion": "2018-09-01-preview",
+                            "name": "[concat(variables('dlsName'), '/Microsoft.Authorization/',  guid(concat(resourceGroup().id, '/', parameters('storageBlobDataContributorID'), '/', variables('workspaceName'), '/', parameters('storageRoleUniqueId'))))]",
+                            "location": "[parameters('location')]",
+                            "properties": {
+                                "roleDefinitionId": "[resourceId('Microsoft.Authorization/roleDefinitions', parameters('storageBlobDataContributorID'))]",
+                                "principalId": "[reference(resourceId('Microsoft.Synapse/workspaces/', variables('workspaceName')), '2019-06-01-preview', 'Full').identity.principalId]",
+                                "principalType": "ServicePrincipal"
+                            }
+                        }
+                    ]
+                }
+            }
+        },
+        {
+            "condition": true,
+            "apiVersion": "2019-05-01",
+            "name": "storageRoleUserDeploymentResource",
+            "type": "Microsoft.Resources/deployments",
+            "subscriptionId": "[subscription().subscriptionId]",
+            "resourceGroup": "[resourceGroup().name]",
+            "dependsOn": [
+                "[variables('workspaceName')]"
+            ],
+            "properties": {
+                "mode": "Incremental",
+                "template": {
+                    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+                    "contentVersion": "1.0.0.0",
+                    "resources": [
+                        {
+                            "type": "Microsoft.Storage/storageAccounts/providers/roleAssignments",
+                            "apiVersion": "2018-09-01-preview",
+                            "name": "[concat(variables('dlsName'), '/Microsoft.Authorization/',  guid(concat(resourceGroup().id, '/', parameters('storageBlobDataContributorID'), '/', parameters('userObjectID'), '/', parameters('storageRoleUniqueId'))))]",
+                            "location": "[parameters('location')]",
+                            "properties": {
+                                "roleDefinitionId": "[resourceId('Microsoft.Authorization/roleDefinitions', parameters('storageBlobDataContributorID'))]",
+                                "principalId": "[parameters('userObjectID')]",
```

Comment:
if the intent here is that the user deploying is granted access, it's not needed... the user will be the owner of the resource